### PR TITLE
Add logging for contact information.

### PR DESCRIPTION
**Pull request by YunfeiBai**

### CODE REVIEW

File: examples/SharedMemory/PhysicsClientC_API.cpp [hunk 1]
```diff
@@ -2659,6 +2659,58 @@ int b3StateLoggingAddLoggingObjectUniqueId(b3SharedMemoryCommandHandle commandHa
...
+	b3Assert(command->m_type = CMD_STATE_LOGGING);
+	if (command->m_type == CMD_STATE_LOGGING)
+	{
+		command->m_updateFlags |= STATE_LOGGING_FILTER_LINK_INDEX_B;
+		command->m_stateLoggingArguments.m_linkIndexB = linkIndexB;
+	}
+	return 0;
+}
+
+int b3StateLoggingSetBodyIndexA(b3SharedMemoryCommandHandle commandHandle, int bodyIndexA)
```
Comment (erwincoumans) [inline examples/SharedMemory/PhysicsClientC_API.cpp]: This is better called b3StateLoggingSetBodyAUniqueId (and b3StateLoggingSetBodyBUniqueId) This is a body unique id, not an index. The links are stored as indices in range [0..numLinks/Joints), but bodyUniqueIds are really unique identifiers.

File: examples/SharedMemory/PhysicsServerCommandProcessor.cpp [hunk 2]
```diff
@@ -950,6 +950,169 @@ struct GenericRobotStateLogger : public InternalStateLogger
...
+	
+	std::string m_fileName;
+	FILE* m_logFileHandle;
+	std::string m_structTypes;
+	btMultiBodyDynamicsWorld* m_dynamicsWorld;
+	bool m_filterLinkA;
+	bool m_filterLinkB;
+	int m_linkIndexA;
+	int m_linkIndexB;
+	int m_bodyIndexA;
```
Comment (erwincoumans) [inline examples/SharedMemory/PhysicsServerCommandProcessor.cpp]: similar as before: m_bodyUniqueId

File: examples/SharedMemory/SharedMemoryCommands.h [hunk 3]
```diff
@@ -628,7 +628,11 @@ enum eStateLoggingEnums
 	STATE_LOGGING_START_LOG=1,
 	STATE_LOGGING_STOP_LOG=2,
 	STATE_LOGGING_FILTER_OBJECT_UNIQUE_ID=4,
-	STATE_LOGGING_MAX_LOG_DOF=8
+	STATE_LOGGING_MAX_LOG_DOF=8,
+	STATE_LOGGING_FILTER_LINK_INDEX_A=16,
+	STATE_LOGGING_FILTER_LINK_INDEX_B=32,
+	STATE_LOGGING_FILTER_BODY_INDEX_A=64,
```
Comment (erwincoumans) [inline examples/SharedMemory/SharedMemoryCommands.h]: BODY_UNIQUE_ID

File: examples/pybullet/pybullet.c [hunk 4]
```diff
@@ -2678,12 +2678,16 @@ static PyObject* pybullet_startStateLogging(PyObject* self, PyObject* args, PyOb
 	char* fileName = 0;
 	PyObject* objectUniqueIdsObj = 0;
 	int maxLogDof = -1;
+	int bodyIndexA = -1;
+	int bodyIndexB = -1;
+	int linkIndexA = -2;
+	int linkIndexB = -2;
 
-	static char* kwlist[] = {"loggingType", "fileName", "objectUniqueIds", "maxLogDof", "physicsClientId", NULL};
+	static char* kwlist[] = {"loggingType", "fileName", "objectUniqueIds", "maxLogDof", "bodyIndexA", "bodyIndexB", "linkIndexA", "linkIndexB", "physicsClientId", NULL};
```
Comment (erwincoumans) [inline examples/pybullet/pybullet.c]: bodyUniqueId

**Comment by erwincoumans** [comment]
Thanks for the contribution. Can you please replace all bodyIndex to bodyUniqueId in your pull request? LinkIndex / JointIndex is ok as-is.

**Comment by erwincoumans** [comment]
Thanks a lot for fixing the naming! I noticed one last left-over in the comments in pybullet, I can fix that also after merging the pull request ("fileName, optional objectUniqueId, maxLogDof, bodyIndexA, bodyIndexB, )

**Comment by YunfeiBai** [comment]
Good catch! Thanks Erwin! I just pushed the fix.

**Comment by erwincoumans** [comment]
Thanks!

**Pull request merged**